contracts/colony/ColonyFunding.sol

```diff
@@ -106,6 +107,84 @@ contract ColonyFunding is
     emit ColonyFundsClaimed(msgSender(), _token, feeToPay, remainder);
   }
 
+  function claimDomainFunds(address _token, uint256 _domainId) public stoppable {
+    require(domainExists(_domainId), "colony-funding-domain-does-not-exist");
+    address domainTokenReceiverAddress = IColonyNetwork(colonyNetworkAddress)
+      .idempotentDeployDomainTokenReceiver(_domainId);
+    uint256 fundingPotId = domains[_domainId].fundingPotId;
+    // It's deployed, so check current balance of pot
+
+    uint256 claimAmount;
+
+    if (_token == address(0x0)) {
+      claimAmount = address(domainTokenReceiverAddress).balance;
+    } else {
+      claimAmount = ERC20Extended(_token).balanceOf(address(domainTokenReceiverAddress));
+    }
+
+    uint256 feeToPay = claimAmount / getRewardInverse(); // ignore-swc-110 . This variable is set when the colony is
+    // initialised to MAX_UINT, and cannot be set to zero via setRewardInverse, so this is a false positive. It *can* be set
+    // to 0 via recovery mode, but a) That's not why MythX is balking here and b) There's only so much we can stop people being
+    // able to do with recovery mode.
+    uint256 remainder = claimAmount - feeToPay;
+    nonRewardPotsTotal[_token] += remainder;
+
+    fundingPots[0].balance[_token] += feeToPay;
+
+    uint256 approvedAmount = domainReputationTokenApprovals[block.chainid][_domainId][_token];
+    if (!tokenEarnsReputationOnPayout(_token) || approvedAmount >= remainder) {
+      // Either the token doesn't earn reputation or there is enough approval
+      // Either way, the domain gets all the funds
+      fundingPots[fundingPotId].balance[_token] += remainder;
+      if (tokenEarnsReputationOnPayout(_token)) {
+        // If it does earn reputation, deduct the approved amount
+        domainReputationTokenApprovals[block.chainid][_domainId][_token] -= remainder;
+      }
+      emit DomainFundsClaimed(msgSender(), _token, _domainId, feeToPay, remainder);
+    } else {
+      // The token earns reputation and there is not enough approvalable
+      // The domain gets what was approved
+      fundingPots[fundingPotId].balance[_token] += approvedAmount;
+      // And the rest goes to the root pot
+      Domain storage rootDomain = domains[1];
+      fundingPots[rootDomain.fundingPotId].balance[_token] += remainder - approvedAmount;
+      domainReputationTokenApprovals[block.chainid][_domainId][_token] = 0;
+      emit DomainFundsClaimed(msgSender(), _token, _domainId, feeToPay, approvedAmount);
+      emit ColonyFundsClaimed(msgSender(), _token, 0, remainder - approvedAmount);
+    }
+
+    // Claim funds
+
+    DomainTokenReceiver(domainTokenReceiverAddress).transferToColony(_token);
+  }
+
+  function tokenEarnsReputationOnPayout(address _token) internal view returns (bool) {
```

Comment:
@arrenv do you want to weigh in on this? The idea is that instead of approving a number of tokens to be sent to the domain, you approve an amount of reputation. The effect is that for tokens with only a small reputation factor (i.e. 10% or so) the approval goes further. It also means that for the user, approvals are always framed in terms of "how much reputation do I want to be able to send to this domain" regardless of the rate.

Reply:
Yeah, we discussed this in the product call yesterday, and they agree it should be a reputation limit.

Reply:
Once we add support for more reputation-bearing tokens are added this will need to be re-implemented along the lines of

```
reputationRemainder = wmul(remainder, reputationRate[token])

...
```